Lib/typing.py

```diff
@@ -1401,7 +1401,25 @@ def _is_callable_members_only(cls):
 
 
 def _no_init(self, *args, **kwargs):
-    raise TypeError('Protocols cannot be instantiated')
+    cls = type(self)
+
+    if cls._is_protocol:
+        raise TypeError('Protocols cannot be instantiated')
+
+    # set correct __init__ method on a first initialization
```

Comment:
Wow the fix is really smart!
I was kind of confused, so could you please change the comments to this? (After renaming `_no_init` to something better, see the other comment).
```
    # Initially, `__init__` of a protocol subclass is set to `_no_init`.
    # The first instantiation of the subclass will call `_no_init` which
    # searches for a proper new `__init__` in the MRO. The new `__init__`
    # replaces the subclass' old `__init__` (ie `_no_init`). Subsequent
    # instantiation of the protocol subclass will thus use the new
    # `__init__` and no longer call `_no_init`.
```

Reply:
Thanks for such a good comment, it will definitely make things more clean for other maintainers.